gprofiler/profilers/php.py

```diff
@@ -66,9 +67,12 @@ def __init__(
         profile_spawned_processes: bool,
         php_process_filter: str,
         php_mode: str,
+        container_names_client: Optional[ContainerNamesClient],
     ):
         assert php_mode == "phpspy", "PHP profiler should not be initialized, wrong php_mode value given"
-        super().__init__(frequency, duration, stop_event, storage_dir, insert_dso_name, profiling_mode)
+        super().__init__(
+            frequency, duration, stop_event, storage_dir, insert_dso_name, profiling_mode, container_names_client
+        )
```

Comment:
I don't like it much that we have to make this change in all profilers.

Logically - there are profiler parameters which are truly parameters per profiler and might change per profiler - `frequency`, `duration`. All others (and specifically `stop_event`, `storage_dir`, `containers_name_client`) are actually passed because we cannot pass "general state parameters" in another way.

Let's refactor it (in a different PR) and then base this PR on that refactor.

* One option - profilers can get the `GProfiler` object and access the settings configured on it via accessors `.get_storage_dir()` etc.
* We can create another object `ProfilerState`, create it in `GProfiler` and pass it in the c'tor

My point is, adding a new "state parameter" like `containers_name_client` should not induce diff in all classes of all c'tors.

Let's open another PR for this refactor please